x-pack/platform/test/functional/page_objects/onechat_page.ts

```diff
@@ -356,4 +359,227 @@ export class OneChatPageObject extends FtrService {
     await this.confirmModalConfirm();
     await this.testSubjects.click('toastCloseButton');
   }
+
+  /*
+   * ==========================
+   * Agents: creation
+   * ==========================
+   */
+  async createAgentViaUI({ id, name, labels }: { id: string; name: string; labels: string[] }) {
+    await this.navigateToApp('agents/new');
+    const selectors = {
+      inputs: {
+        id: 'agentSettingsIdInput',
+        displayName: 'agentSettingsDisplayNameInput',
+        description: 'agentSettingsDescriptionInput',
+        labels: 'comboBoxSearchInput',
+      },
+      saveButton: 'agentFormSaveButton',
+      labelsComboBox: 'agentSettingsLabelsComboBox',
+    };
+    await this.testSubjects.append(selectors.inputs.id, id);
+    await this.testSubjects.append(selectors.inputs.displayName, name);
+    await this.testSubjects.append(selectors.inputs.description, `Agent for testing ${id}`);
+    const labelsComboBox = await this.testSubjects.find(selectors.labelsComboBox);
+    const labelsInput = await this.testSubjects.findDescendant(
+      selectors.inputs.labels,
+      labelsComboBox
+    );
+    for (const label of labels) {
+      await labelsInput.click();
+      await labelsInput.type(label);
+      await labelsInput.pressKeys(this.browser.keys.ENTER);
+    }
+    await this.testSubjects.click(selectors.saveButton);
+  }
+
+  /*
+   * ==========================
+   * Agents: list page
+   * ==========================
+   */
+  agentsListSearch() {
+    const contentSelector = subj('agentBuilderAgentsListContent');
+    const searchSelector = `${contentSelector} .euiFieldSearch`;
+    const clearButtonSelector = `${contentSelector} ${subj('clearSearchButton')}`;
+    const findSearch = () => this.find.byCssSelector(searchSelector);
+    return {
+      type: async (term: string) => {
+        const search = await findSearch();
+        await search.click();
+        await search.type(term);
+      },
+      clear: async () => {
+        await this.find.clickByCssSelector(clearButtonSelector);
+        const search = await findSearch();
+        expect(await search.getAttribute('value')).to.be('');
```

Comment:
I think we should avoid having assertions in this file and keep the assertions in the test file itself. This was the feedback Pierre gave me when I initially created this file and added these helper methods.